invenio_circulation/permissions.py

```diff
@@ -17,17 +17,19 @@
 from invenio_access.permissions import Permission
 from invenio_records_rest.utils import allow_all
 
+from invenio_circulation.errors import InvalidPermission
+
 loan_access = action_factory('loan-read-access')
 
 
-def check_permission(permission):
+def check_permission(action, permission):
     """Abort if permission is not allowed.
 
     :param permission: The permission to check.
     """
     if permission is not None and not permission.can():
         if current_user.is_authenticated:
-            abort(403, 'You do not have a permission for this action')
+            raise InvalidPermission(action=action, permission=permission)
```

Comment:
I would remove this one, since it is a generic Invenio exception, otherwise the circulation module will behave differently for REST Api permissions checks.
I would also **NOT** expose the action name.

Reply:
Meaning remove it completely or revert it to `abort(403)`